### ISSUE TITLE
`temperatured_softmax` normalises over the whole array; float128 fallback can't run on arm64

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#13]
Two problems in `temperatured_softmax` (`rhythmic_relationships/evaluate.py:19-29`).

**1. The normalisation has no `axis`.**

```python
probs = np.exp(logits / temperature) / np.sum(np.exp(logits / temperature))
```

For `(B, V)` logits this divides by the sum over the *whole array*, so rows don't sum to 1:

```
logits = [[1,2,3],[1,1,1]]  ->  row sums = [0.787, 0.213]
```

`torch.multinomial` renormalises per row so sampling survives, but `nucleus(probs[j], p)` and the `entropy(probs)` logging in `tvae/train.py` and `hits_dec_xformer/train.py` are computed on a mis-normalised distribution. Should be `np.sum(..., axis=-1, keepdims=True)`.

**2. No max subtraction, and the overflow fallback can't run on this platform.**

```python
except:
    print("Overflow detected, use 128-bit")
    logits = logits.astype(np.float128)
```

`np.float128` does not exist on arm64 macOS, so the fallback raises `AttributeError` from inside the `except` — on the machine this repo appears to be developed on:

```
np.float128 available: False | numpy 2.5.1 | platform arm64
```

**Fix**: subtract the row max before exponentiating (`logits -= logits.max(axis=-1, keepdims=True)`), which makes the overflow path unnecessary, and delete the bare `except:` + float128 branch.
